trino/Dockerfile

```diff
@@ -4,56 +4,62 @@
 # hadolint global ignore=DL3038,DL4006
 
-FROM stackable/image/java-base
+FROM stackable/image/java-devel AS storage-connector-builder
 
-ARG PRODUCT
-# External OPA authorizer only required for 414 and 428, included in 438
-# The `OPA_AUTHORIZER` is not set in the conf.py for version 438 to remove the ARG as soon as 414 and 428 are removed
-ARG OPA_AUTHORIZER
-ARG JMX_EXPORTER
-ARG RELEASE
 ARG STORAGE_CONNECTOR
+ARG STORAGE_CONNECTOR_GIT_REV
 
-LABEL name="Trino" \
-      maintainer="info@stackable.tech" \
-      vendor="Stackable GmbH" \
-      version="${PRODUCT}" \
-      release="${RELEASE}" \
-      summary="The Stackable image for Trino." \
-      description="This image is deployed by the Stackable Operator for Trino."
+WORKDIR /stackable
 
-RUN microdnf update && \
-    microdnf install \
-    gzip \
-    httpd-tools \
-    python \
-    tar \
-    zip && \
-    microdnf clean all && \
-    rm -rf /var/cache/yum
+RUN git clone https://github.com/snowlift/trino-storage --branch=v${STORAGE_CONNECTOR} --depth=1 && \
+    cd trino-storage && git checkout ${STORAGE_CONNECTOR_GIT_REV}
+RUN --mount=type=cache,target=/root/.m2/repository cd trino-storage && \
+    # Upstream builds are marked as -SNAPSHOT, even for release builds
+    mvn versions:set -DnewVersion=${STORAGE_CONNECTOR} && \
+    mvn package -DskipTests
+
+# # TODO: remove the OPA authorizer builder once Trino versions 414 and 428 are removed
+FROM stackable/image/java-devel AS opa-authorizer-builder
+
+ARG OPA_AUTHORIZER
+ARG OPA_AUTHORIZER_GIT_REV
 
-USER stackable
 WORKDIR /stackable
 
-COPY --chown=stackable:stackable trino/stackable /stackable
-COPY --chown=stackable:stackable trino/licenses /licenses
+RUN if [[ -n "${OPA_AUTHORIZER}" ]]; then \
+        # Need to use deep clone because trino-opa-authorizer doesn't publish tags
+        git clone https://github.com/stackabletech/trino-opa-authorizer && \
+        cd trino-opa-authorizer && git checkout ${OPA_AUTHORIZER_GIT_REV}; \
+    fi
+RUN --mount=type=cache,target=/root/.m2/repository if [[ -n "${OPA_AUTHORIZER}" ]]; then \
+        cd trino-opa-authorizer && \
+        mvn package -DskipTests; \
+    fi
 
-RUN curl --fail -L https://repo.stackable.tech/repository/packages/trino-server/trino-server-${PRODUCT}.tar.gz | tar -xzC . && \
-    ln -s /stackable/trino-server-${PRODUCT} /stackable/trino-server
+# Create an empty dummy plugin folder for newer versions that don't need the plugin anymore
+RUN mkdir -p trino-opa-authorizer/target/out
 
-RUN curl --fail https://repo.stackable.tech/repository/packages/jmx-exporter/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar \
-    -o /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar && \
-    chmod +x /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar && \
-    ln -s /stackable/jmx/jmx_prometheus_javaagent-${JMX_EXPORTER}.jar /stackable/jmx/jmx_prometheus_javaagent.jar
+FROM stackable/image/java-devel AS builder
```

Comment:
```suggestion
# Not tagging base image because it is built as part of the same process
# hadolint ignore=DL3006
FROM stackable/image/java-devel AS builder
```